### PR TITLE
Feature/rnp 111

### PR DESCRIPTION
**Pull request by MichielVrins**

### CODE REVIEW

File: android/src/main/java/com/onegini/mobile/sdk/reactnative/RNOneginiSdk.kt [hunk 1]
```diff
@@ -282,21 +276,8 @@ class RNOneginiSdk(private val reactContext: ReactApplicationContext) : ReactCon
...
-                        }
-
-                        override fun onError(error: OneginiAppToWebSingleSignOnError) {
-                            promise.reject(error.errorType.toString(), error.message)
-                        }
-                    }
-                )
-            }
+            null -> promise.rejectWithNullError(Uri.paramName, Uri.type)
+            else -> sdkWrapper.startSingleSignOn(uri, promise)
```
Comment (PiotrKlis) [inline android/src/main/java/com/onegini/mobile/sdk/reactnative/RNOneginiSdk.kt:280]: with logic extracted into separate class, I think we could go with classic elvis operator here
```
uri?.let { sdkWrapper.startSingleSignOn(uri, promise) } ?: promise.rejectWithNullError(Uri.paramName, Uri.type)
```

File: android/src/test/java/com/onegini/mobile/sdk/reactnative/OneginiSdkWrapperTests.kt [hunk 2]
```diff
@@ -88,20 +91,24 @@ class OneginiSdkWrapperTests {
         every { Arguments.createArray() } answers { JavaOnlyArray() }
         every { Arguments.createMap() } answers { JavaOnlyMap() }
 
+
+
```
Comment (PiotrKlis) [inline android/src/test/java/com/onegini/mobile/sdk/reactnative/OneginiSdkWrapperTests.kt:95]: ✂️ redundant empty space ✂️

File: android/src/test/java/com/onegini/mobile/sdk/reactnative/OneginiSdkWrapperTests.kt [hunk 3]
```diff
@@ -181,4 +188,11 @@ class OneginiSdkWrapperTests {
 
         verify(authenticateUserUseCase).invoke("123456", "1", promiseMock)
     }
+
+    @Test
+    fun `when startSingleSignOn method is called calls startSingleSignOnUseCase with proper params`() {
```
Comment (PiotrKlis) [inline android/src/test/java/com/onegini/mobile/sdk/reactnative/OneginiSdkWrapperTests.kt:193]: When startSingleSignOn method is called, Then startSingleSignOnUseCase is called with proper params

File: android/src/test/java/com/onegini/mobile/sdk/reactnative/StartSingleSignOnUseCaseTests.kt [hunk 4]
```diff
@@ -0,0 +1,95 @@
...
+    @get:Rule
+    val reactArgumentsTestRule = ReactArgumentsTestRule()
+
+    @Mock(answer = Answers.RETURNS_DEEP_STUBS)
+    private lateinit var oneginiSdk: OneginiSDK
+
+    @Mock
+    private lateinit var uriFacade: UriFacade
+
+    // We need to deep stub here to mock a uri object's .toString() as we cant pass in a uriFacade
```
Comment (PiotrKlis) [inline android/src/test/java/com/onegini/mobile/sdk/reactnative/StartSingleSignOnUseCaseTests.kt:37]: as we cant pass `it into` uriFacade
isn't it better? 🤔

**Review by PiotrKlis — CHANGES_REQUESTED** [review]
left few minor comments

File: android/src/test/java/com/onegini/mobile/sdk/reactnative/OneginiSdkWrapperTests.kt (hunk 3 above)
Comment (MichielVrins) [reply android/src/test/java/com/onegini/mobile/sdk/reactnative/OneginiSdkWrapperTests.kt:193]: Ignore this one as this useless test file has been removed in another commit.

File: android/src/test/java/com/onegini/mobile/sdk/reactnative/StartSingleSignOnUseCaseTests.kt (hunk 4 above)
Comment (MichielVrins) [reply android/src/test/java/com/onegini/mobile/sdk/reactnative/StartSingleSignOnUseCaseTests.kt]: It actually means something else so I have adjusted it to hopefully make it more understandable.

File: android/src/test/java/com/onegini/mobile/sdk/reactnative/StartSingleSignOnUseCaseTests.kt [hunk 5]
```diff
@@ -0,0 +1,94 @@
...
+    private val mockedTokenString = "mockedToken"
+    private val mockedRedirectUrlString = "mockedRedirectUrl"
+
+    @Before
+    fun setup() {
+        startSingleSignOnUseCase = StartSingleSignOnUseCase(oneginiSdk, uriFacade)
+    }
+
+    @Test
+    fun `When oginini getAppToWebSingleSignOn calls onSuccess on the handler, promise should resolve with a map containing the content from the result`() {
```
Comment (PiotrKlis) [inline android/src/test/java/com/onegini/mobile/sdk/reactnative/StartSingleSignOnUseCaseTests.kt:62]: Add `then` after coma to follow the pattern.

File: android/src/test/java/com/onegini/mobile/sdk/reactnative/StartSingleSignOnUseCaseTests.kt [hunk 6]
```diff
@@ -0,0 +1,94 @@
...
+        startSingleSignOnUseCase(correctUri, promiseMock)
+        argumentCaptor<JavaOnlyMap> {
+            verify(promiseMock).resolve(capture())
+            assertEquals(firstValue.getString("url"), mockedRedirectUrlString)
+            assertEquals(firstValue.getString("token"), mockedTokenString)
+        }
+    }
+
+    @Test
+    fun `When oginini getAppToWebSingleSignOn calls onError on the handler, promise should reject with the error message and code`() {
```
Comment (PiotrKlis) [inline android/src/test/java/com/onegini/mobile/sdk/reactnative/StartSingleSignOnUseCaseTests.kt:77]: Add then after coma to follow the pattern from different test classes.

**Review by PiotrKlis — APPROVED** [review]

**Pull request merged**